tests/rados/test_mon_connection_scores.py

```diff
@@ -0,0 +1,246 @@
+            raise Exception("mon service not removed error")
+
+        if not connection_score_checks(
+            mon_nodes, rados_obj, mon_nums - 1, mon_election_obj
+        ):
+            raise Exception("Connection score checks failed")
+
+        # Restart monitor service and perform connection
+        # score validation
+        mon_host = random.choice(mon_nodes)
+
+        if not rados_obj.change_daemon_systemctl_state(
+            action="restart", daemon_type="mon", daemon_id=mon_host.hostname
+        ):
+            log.error(f"Failed to restart mon on host {mon_host.hostname}")
+            raise Exception("Mon restart failure error")
+
+        if not connection_score_checks(
+            mon_nodes, rados_obj, mon_nums, mon_election_obj
+        ):
+            raise Exception("Connection score checks failed")
+
+        # Stop monitor service and perform connection score validation
+        if not rados_obj.change_daemon_systemctl_state(
+            action="stop", daemon_type="mon", daemon_id=mon_host.hostname
+        ):
+            log.error(f"Failed to stop mon on host {mon_host.hostname}")
+            raise Exception("Mon stop failure error")
+
+        if not connection_score_checks(
+            mon_nodes, rados_obj, mon_nums, mon_election_obj
+        ):
+            raise Exception("Connection score checks failed")
+
+        # Start stopped monitor service and perform connection score validation
+        if not rados_obj.change_daemon_systemctl_state(
+            action="start", daemon_type="mon", daemon_id=mon_host.hostname
+        ):
+            log.error(f"Failed to start mon on host {mon_host.hostname}")
+            raise Exception("Mon start failure error")
+
+        if not connection_score_checks(
+            mon_nodes, rados_obj, mon_nums, mon_election_obj
+        ):
+            raise Exception("Connection score checks failed")
+
+    except Exception as e:
+        log.error(f"Failed with exception: {e.__doc__}")
+        log.exception(e)
+        return 1
+
+    finally:
+        log.info(
+            "\n \n ************** Execution of finally block begins here *************** \n \n"
+        )
+        # Setting the mon service as managed by cephadm
+        if not mon_workflow_obj.set_mon_service_managed_type(unmanaged=False):
+            log.error("Could not set the mon service to managed")
+            return 1
+
+        # log cluster health
+        rados_obj.log_cluster_health()
+
+        # check for crashes after test execution
+        if rados_obj.check_crash_status():
+            log.error("Test failed due to crash at the end of test")
+            return 1
+
+    log.info(
+        f"Successfully verified connection scores at different scenarios\
+         later reverted cluster to original state"
+    )
+    return 0
+
+def connection_score_checks(mon_nodes, rados_obj, num_mons, mon_election_obj) -> bool:
```

Comment:
Move the method to mon workflows

Reply:
Moved connection_score_checks() to ceph/rados/monitor_workflows.py